src/lib/arch/aarch64/Instruction_decode.cc

```diff
@@ -372,17 +372,21 @@ void Instruction::decode() {
     isLoad_ = true;
   }
 
-  if ((227 < metadata.opcode && metadata.opcode < 254) ||    // AND
-      (299 < metadata.opcode && metadata.opcode < 321) ||    // BIC
-      (733 < metadata.opcode && metadata.opcode < 759) ||    // EOR/EON
-      (2626 < metadata.opcode && metadata.opcode < 2655)) {  // ORR/ORN
+  if ((264 <= metadata.opcode && metadata.opcode <= 267) ||    // AND
+      (1063 <= metadata.opcode && metadata.opcode <= 1084) ||  // AND (pt.2)
+      (284 <= metadata.opcode && metadata.opcode <= 287) ||    // BIC
+      (1167 <= metadata.opcode && metadata.opcode <= 1183) ||  // BIC (pt.2)
+      (321 <= metadata.opcode && metadata.opcode <= 324) ||    // EOR/EON
+      (1707 <= metadata.opcode && metadata.opcode <= 1736) ||  // EOR/EON (pt.2)
+      (771 <= metadata.opcode && metadata.opcode <= 774) ||    // ORR/ORN
+      (3748 <= metadata.opcode && metadata.opcode <= 3771)) {  // ORR/ORN (pt.2)
```

Comment:
It would do yes, however due to the long enum names and clang formatting performed it will likely lead to each case being multi-line.... so there is a trade-off here. @rahahahat @jj16791 any input on this?

Reply:
Given that the logic has comments identifying the command/class of commands, IMO it is fine as long these opcodes are raw numbers that only exist within the scope of this file and If there is appropriate documentation which can help identify these opcodes for e.g link to the ISA etc.